### PR TITLE
Update golang.org/x/crypto for v2.8.x

### PR DESCRIPTION
**Pull request by na--**
**What this PR does / why we need it**:

Updates the `golang.org/x/crypto` dependency to its latest version, `v0.18.0`

**Which issue(s) this PR fixes**:

Resolves https://nvd.nist.gov/vuln/detail/CVE-2023-48795

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Tests updated
- [x] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label

### CODE REVIEW

**Comment by na--** [comment]
The `CircleCI Pipeline` check is failing, but from what I can see we no longer use CircleCI, so I assume it was used at the time the `release-v2.8.x` branch was split off from `main`? :thinking:

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,10 @@
 * [11606](https://github.com/grafana/loki/pull/11606) **dannykopping** Fixed regression adding newlines to HTTP error response bodies which may break client integrations.
 * [11657](https://github.com/grafana/loki/pull/11657) **ashwanthgoli** Log results cache: compose empty response based on the request being served to avoid returning incorrect limit or direction.
 
+##### Security
+
+* [11743](https://github.com/grafana/loki/pull/11743) Update `https://github.com/grafana/loki/pull/11743` to `v0.18.0`
```
Comment (poyzannur) [inline CHANGELOG.md:14]:
```suggestion
* [11743](https://github.com/grafana/loki/pull/11743) Update `golang.org/x/crypto` to `v0.18.0`
```

Comment (na--) [reply CHANGELOG.md:14]: :man_facepalming: thanks!

**Comment by na--** [comment]
Good points, will fix shortly!

**Review by poyzannur — APPROVED** [review]

**Pull request merged**